### PR TITLE
Decouple severity from the cacheability state

### PR DESCRIPTION
**Pull request by melix**
The existing validation code conflates the severity of a problem with
the cacheability state. This commit decouples things by making the
severity only talk about severity, and having the cacheability state
carried as part of the problem payload.

The enum value is still used for all problems which do not use the
new builder mechanism. The plan is to finally get rid of this enum
value.

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test QFL

File: subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/AbstractValidationProblemBuilder.java [hunk 1]
```diff
@@ -86,6 +87,12 @@ public T addPossibleSolution(Supplier<String> solution, Action<? super SolutionB
         return Cast.uncheckedCast(this);
     }
 
+    @Override
+    public T isCachabilityProblemOnly() {
```
Comment (wolfs) [inline subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/AbstractValidationProblemBuilder.java:91]:
```suggestion
    public T onlyAffectsCacheableWork() {
```

This method sounds very much like a getter, though it is something different. I think `cacheabilityProblemOnly()` or `onlyAffectsCacheableWork()` would be better. If you want to keep this method, it is missing an `e`.

Comment (melix) [reply subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/AbstractValidationProblemBuilder.java:91]: I had the same comment to myself after using it in the next PR :D

**Comment by melix** [comment]
@bot-gradle test QFL

**Comment by melix** [comment]
@bot-gradle test QFL

**Comment by melix** [comment]
@bot-gradle test QFL

**Pull request merged**